File: back/DevC/src/main/java/com/back/devc/domain/post/comment/dto/CommentResponse.java
package com.back.devc.domain.post.comment.dto;

import com.back.devc.domain.post.comment.attachment.dto.CommentAttachmentResponse;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public record CommentResponse(
        Long commentId,
        Long postId,
        String postTitle,
        Long userId,
        String nickname,
        Long parentCommentId,
        String content,
        boolean isDeleted,
        LocalDateTime createdAt,
        LocalDateTime updatedAt,
        List<CommentResponse> replies,
        List<CommentAttachmentResponse> attachments
) {

    public static CommentResponse of(
            Long commentId,
            Long postId,
            String postTitle,
            Long userId,
            String nickname,
            Long parentCommentId,
            String content,
            boolean isDeleted,
            LocalDateTime createdAt,
            LocalDateTime updatedAt
    ) {
        return new CommentResponse(
                commentId,
                postId,
                postTitle,
                userId,
                nickname,
                parentCommentId,
                content,
                isDeleted,
                createdAt,
                updatedAt,
                new ArrayList<>(),
                new ArrayList<>()
        );
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/post/comment/dto/CommentResponse.kt
package com.back.devc.domain.post.comment.dto

import com.back.devc.domain.post.comment.attachment.dto.CommentAttachmentResponse
import java.time.LocalDateTime

data class CommentResponse(
    val commentId: Long,
    val postId: Long,
    val postTitle: String,
    val userId: Long,
    val nickname: String,
    val parentCommentId: Long?,
    val content: String,
    val isDeleted: Boolean,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val replies: MutableList<CommentResponse> = mutableListOf(),
    val attachments: MutableList<CommentAttachmentResponse> = mutableListOf(),
) {
    companion object {
        @JvmStatic
        fun of(
            commentId: Long,
            postId: Long,
            postTitle: String,
            userId: Long,
            nickname: String,
            parentCommentId: Long?,
            content: String,
            isDeleted: Boolean,
            createdAt: LocalDateTime,
            updatedAt: LocalDateTime,
        ): CommentResponse {
            return CommentResponse(
                commentId = commentId,
                postId = postId,
                postTitle = postTitle,
                userId = userId,
                nickname = nickname,
                parentCommentId = parentCommentId,
                content = content,
                isDeleted = isDeleted,
                createdAt = createdAt,
                updatedAt = updatedAt,
            )
        }
    }
}